agent/api/task/task_attachment_handler.go

```diff
@@ -120,8 +121,17 @@ func handleTaskAttachments(acsTask *ecsacs.Task, task *Task) error {
 					Type:   apiresource.EBSTaskAttach,
 					Volume: ebs,
 				}
+				ebsVolumes[ebs.VolumeName] = true
 				task.Volumes = append(task.Volumes, taskVolume)
 			}
+			// We're removing all incorrect volume configuration that were intially passed over from ACS
```

Comment:
nit: "all incorrect volume configurations"

Reply:
We are saying that the task payload we get from ECS backend is incorrect? That reads very odd to me. Is there a better way to put it?

Reply:
er sorry perhaps we can say "incomplete" here

Reply:
Even incomplete sounds wrong... non-blocking but let's rethink to prevent future confusion.

Reply:
completely agree